src/app/perspective-status.component.ts

```diff
@@ -236,34 +242,39 @@ export class PerspectiveStatus implements OnChanges, AfterViewInit, AfterViewChe
    * callback, to ensure that the ViewChild has been updated.
    */
   ngAfterViewChecked() {
-    // TODO(rachelrosen): Queue up all animations from ngOnChanges and put them
-    // in a proper timeline (instead of just chaining promises), so state
-    // doesn't accidentally get messed up. This is not a production concern
-    // because state will not be changing there; it's only a concern for changing
-    // settings with the customizable form.
-    let updatesToDo = Promise.resolve();
-    if (this.scoreThresholdsChanged) {
-      this.scoreThresholdsChanged = false;
-      // Skip the updateWidgetStateAnimation if the loading style also changed,
-      // since the loading style update takes care of the widget state change
-      // animation.
-      if (!this.loadingIconStyleChanged) {
-        updatesToDo.then(() => {
+    if (this.scoreThresholdsChanged || this.loadingIconStyleChanged) {
+      let afterChangesTimeline = new TimelineMax({
+        onStart: () => {
+          this.ngZone.run(() => {
+            console.debug('Started animations in ngAfterViewChecked');
+          });
+        },
+        onComplete: () => {
+          this.ngZone.run(() => {
+            console.debug('Completing animations in ngAfterViewChecked');
+          });
+        }
+      });
+      // Run in a Promise resolve statement so we don't get an
+      // ExpressionChangedAfterItHasBeenCheckedError.
+      Promise.resolve().then(() => {
+        if (this.loadingIconStyleChanged) {
+          this.loadingIconStyleChanged = false;
+          console.debug('Setting loadingIconStyleChanged to false');
+          afterChangesTimeline.add(this.getUpdateWidgetElementAnimation());
+        } else if (this.scoreThresholdsChanged) {
+          console.debug('Setting scoreThresholdsChanged to false');
```

Comment:
super-nitty nit: logging and setting statement order is reversed in this and the above block.

Reply:
Done